sklearn/preprocessing/_discretization.py

```diff
@@ -102,6 +103,11 @@ class KBinsDiscretizer(BaseEstimator, TransformerMixin):
     :class:`sklearn.compose.ColumnTransformer` if you only want to preprocess
     part of the features.
 
+    ``KBinsDiscretizer`` might produce constant features (e.g., when
+    ``encode = 'onehot'`` and certain bins do not contain any data).
```

Comment:
Maybe mention strategy=uniform here

Reply:
> Maybe mention strategy=uniform here

Why? Other strategies also suffer from this problem.

Reply:
Do they? Since the others follow the empirical distribution of the feature, and remove empty bins how can they have no data at training time?

Reply:
> Do they? Since the others follow the empirical distribution of the feature, and remove empty bins how can they have no data at training time?

We only remove bins whose width <= 0, e.g., 
```
from sklearn.preprocessing import KBinsDiscretizer
X = [[1], [2]]
kb = KBinsDiscretizer(n_bins=5, encode='ordinal')
kb.fit_transform(X)
# array([[0.], [4.]])
```
Maybe I should use ``bins whose width <= 0`` instead of ``Redundant bins (i.e., bins whose width <= 0)`` since it's difficult to define redundant bins?

Reply:
Ahh... Okay. A bit of an edge case since there are fewer samples than bins which maybe we should prohibit anyway